File: ios/ReactNativeHashcashModule.swift
import ExpoModulesCore

public class ReactNativeHashcashModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ReactNativeHashcash")


//    AsyncFunction("calculateHashcash") { (k: UInt, identifier: String, promise: Promise) in
//      Task {
//        promise.resolve(await calculateHashcashTask(k: k, identifier: identifier))
//      }
//    }
      
    AsyncFunction("calculateHashcash") { (k: UInt, identifier: String) in
        calculateHashcashQueue(k: k, identifier: identifier)
    }

    // AsyncFunction("calculateHashcash") { (k: UInt, identifier: String) in
    //   calculateHashcashSequential(k: k, identifier: identifier)
    // }
  }
}

File: android/src/main/java/expo/modules/hashcashold/ReactNativeHashcashModule.kt
package expo.modules.hashcashold

import expo.modules.kotlin.Promise
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ReactNativeHashcashModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ReactNativeHashcash")

//    AsyncFunction("calculateHashcash") { k: Int, identifier: String, promise: Promise ->
//      promise.resolve(calculateHashcashSequential(k.toUInt(), identifier))
//    }

    AsyncFunction("calculateHashcash") { k: Int, identifier: String, promise: Promise ->
      promise.resolve(calculateHashcashMulti(k.toUInt(), identifier))
    }
  }
}
